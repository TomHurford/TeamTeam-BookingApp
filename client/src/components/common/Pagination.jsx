import React from "react";
import _ from "lodash";
import PropTypes from 'prop-types';

const Pagination = (props) => {
  const {
    itemsCount: elementCount,
    pageSize,
    currentPage,
    onPageChange,
  } = props;
  const pageCount = Math.ceil(elementCount / pageSize);
  if (pageCount === 1) return null;
  const pages = _.range(1, pageCount + 1);

  return (
    <ul className="pagination">
      {pages.map((page) => (
        <a
          href="#/"
          className="page-link"
          onClick={() => onPageChange(page)}
        >
          <li
            key={page}
            className={page === currentPage ? "page-item active" : "page-item"}
          >
            {page}
          </li>
        </a>
      ))}
    </ul>
  );
};

Pagination.propTypes = {
  'itemsCount' : PropTypes.number,
  pageSize : PropTypes.number,
  currentPage : PropTypes.number,
  onPageChange : PropTypes.func
}; 
export default Pagination;
