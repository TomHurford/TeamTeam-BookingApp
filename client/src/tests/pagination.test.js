import React from 'react';
import renderer from 'react-test-renderer';
import Pagination from '../components/common/Pagination';

test('Pagination snapshot test', () => {
  const component = renderer.create(<Pagination />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});