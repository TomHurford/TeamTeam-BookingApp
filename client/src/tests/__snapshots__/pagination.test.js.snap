// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination snapshot test 1`] = `
<ul
  className="pagination"
>
  <a
    className="page-link"
    href="#/"
    onClick={[Function]}
  >
    <li
      className="page-item"
    >
      1
    </li>
  </a>
</ul>
`;
